postcss/css-compile-variables.js

```diff
@@ -0,0 +1,87 @@
+import { Color } from "color";
+
+let aliasMap;
+let resolvedMap;
+const RE_VARIABLE_VALUE = /var\((--(.+)--(.+)-(.+))\)/;
+
+function getValueFromAlias(alias) {
+  const derivedVariable = aliasMap.get(`--${alias}`);
+  return resolvedMap.get(derivedVariable); // what if we haven't resolved this variable yet?
+}
+
+function resolveDerivedVariable(decl, variables) {
+  const matches = decl.value.match(RE_VARIABLE_VALUE);
```

Comment:
For things like gradients where we use derived colors it might be useful to use https://www.npmjs.com/package/postcss-value-parser here and walk the node tree looking for var expressions, e.g.:
```css
h1 {
	background: linear-gradient(var(--color1-darker-20), var(--color1--ligher-10));
}
```

Reply:
Example of that here fwiw: https://github.com/devex-web-frontend/postcss-assets-rebase/blob/master/index.js

Reply:
> Example of that here fwiw: https://github.com/devex-web-frontend/postcss-assets-rebase/blob/master/index.js

This is using the outdated postcss syntax though